### DISCUSSION TITLE
Push declined

### DISCUSSION THREAD

**Discussion by vitaldii** [community/community#52687] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hello I wanted to push folder to my GitHub repository and faced with one problem. After push I was told : ! [remote rejected] main -> main (push declined due to email privacy restrictions). I'm feeling really dumb, can someone tell me step by step how to solve the issue.

**Comment by neutrinomuon** [comment, marked as answer]
Hi, @vitaldii,

Don't worry, sometimes it might be confusing. The "push declined due to email privacy restrictions" error message on GitHub typically indicates that the **email address** associated with the user's account is **private**, and Git is unable to access it to complete the push.

This error message is usually **not** related to the use of a token versus a password for authentication (which is another common problem that you can also check). 

Instead, it is caused by a user's account settings. To solve the issue, you should verify their email address with GitHub and make sure it is set to "**public**" in your account settings.

Additionally, if you are pushing changes from a cloned repository, then you should check that their local Git configuration is up-to-date and the configured email address is correct. This can be done by running `git config user.email` in the terminal, and verifying if the email address that pops up is the same as the one associated with their GitHub account.

I hope this helps you, please let me know

**Reply by vitaldii** [reply]
Yeah, it helped me, thank you

**Comment by Suthiwat1** [comment]
เมื่อ พฤ. 13 เม.ย. 2566 เวลา 15:54 Jean Gomes ***@***.***>
เขียนว่า:

**Comment by airtower-luna** [comment]
> Instead, it is caused by a user's account settings. To solve the issue, you should verify their email address with GitHub and make sure it is set to "public" in your account settings.

Or use the right "noreply" address, if they don't want to make their email address public. This will require changing the history to replace the affected author/committer email address. See also: https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address#about-commit-email-addresses

**Reply by neutrinomuon** [reply]
Thanks @airtower-luna, also an important information :)!

**Reply by vitaldii** [reply]
Thank you too for extra info

**Comment by Shoaibkhan0707** [comment]
! [remote rejected] main -> main (push declined due to email privacy restrictions)
error: failed to push some refs to 'https://github.com/Shoaibkhan0707/colleage.git'

**Reply by n8tr0nc** [reply]
need help with this also! anyone?
